packages/platform-ios/src/commands/runIOS/index.js

```diff
@@ -71,55 +71,39 @@ function runIOS(_: Array<string>, ctx: ConfigT, args: FlagsT) {
     }),
   );
 
-  if (args.device) {
-    const selectedDevice = matchingDevice(devices, args.device);
-    if (selectedDevice) {
-      return runOnDevice(
-        selectedDevice,
-        scheme,
-        xcodeProject,
-        args.configuration,
-        args.packager,
-        args.verbose,
-        args.port,
-        args.terminal,
-      );
+  if (!args.device) {
+    if (args.udid) {
+      // $FlowIssue: args.udid is defined in this context
+      return runOnDeviceByUdid(scheme, xcodeProject, devices, args);
     }
-    if (devices && devices.length > 0) {
-      // $FlowIssue: args.device is defined in this context
-      logger.error(`Could not find device with the name: "${args.device}".
+
+    return runOnSimulator(xcodeProject, scheme, args);
+  }
+
+  const selectedDevice = matchingDevice(devices, args.device);
+  if (selectedDevice) {
+    return runOnDevice(selectedDevice, scheme, xcodeProject, args);
+  }
+
+  if (devices && devices.length > 0) {
+    // $FlowIssue: args.device is defined in this context
+    return logger.error(`Could not find device with the name: "${args.device}".
 Choose one of the following:${printFoundDevices(devices)}`);
-    } else {
-      logger.error('No iOS devices connected.');
-    }
-  } else if (args.udid) {
```

Comment:
Any reason to change the order of `if/else` in the file? I quite liked the previous one (was a bit more readable to me)

Reply:
I think this has something to do with the negation, which I am not particularly in favour of. 

Also, how about we remove
```
if (args.udid) {
      // $FlowIssue: args.udid is defined in this context
      return runOnDeviceByUdid(scheme, xcodeProject, devices, args);
    }
```
and do something like:
```
if (args.udid || args.device) {
   const selectedDevice = args.udid ? matchingDeviceByUdid(devices, args.udid) : matchingDevice(devices, args.device);

  // logic
}
```

Benefits - it's explicit and we can handle missing device in one place. Right now, we do it in both places.

Reply:
Also, we should update the type of `device` from `?string` to `?string | ?boolean` because (i've just found out), using `--device` (without a value) will result in a "first available simulator" being picked up.

Sorry if this is annoying, but there's a lot of tiny things here that could be improved that I think we could do while we are here. This is not related to issues with your code per se :)

Reply:
Good points, I'll work on them later on today.

No worries about the review though, it's really nice to improve this part!

If you have any major ideas that don't fit in here you can also create an issue and assign to me, would be happy to work on it.

Reply:
Does it make sense to have `typeof device === Boolean` though? won't it just be `undefined`?

I think having the command in `options` array as `--device [string]` and type as `?string | ?boolean` can be a little strange.

Reply:
@lucasbento, right now, we can either do `--device "Mike's iPhone"` or `--device`. How we can type that? 

I believe, that given the way Commander works, `--device` will result in `args.device` being `true`, which is why it's been handled this way.

Reply:
Maybe `?(string | true)`

Reply:
Alright, I'll do it!